### PR TITLE
Security fixes

### PR DESCRIPTION
**Pull request by praveenKDass**
# Description
These recommendations are intended to promote code quality and team communication during software development. They cover a variety of topics, including ensuring that pull requests are submitted to the correct branch, documenting new methods, preserving consistency across many services, and avoiding typical blunders like accessing APIs or DB queries within loops. Sensitive data should not be uploaded, and changes to environment variables or database models should be executed consistently. Teams may work more effectively and develop higher-quality software by adhering to these standards.

## Type of change
Please choose appropriate options.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Enhancement (additive changes to improve performance)
- [ ]  This change requires a documentation update

## Checklist 

- [ ]  It's critical to avoid making needless file modifications in contributions, such as adding new lines, console logs, or additional spaces, to guarantee cleaner and more efficient code. Furthermore, eliminating unnecessary imports from a file might enhance code readability and efficiency.
- [ ]  Ensure that the pull request is assigned to the right base branch and that the development branch name contains the JIRA Task Id. Furthermore, each commit message should include the JIRA Task Id in the manner "ED-100: message".
- [ ]  Only update packages if it is mentioned and authorized in the design document, and make sure that you have the required permissions.
- [ ]  Avoid making API and database queries inside a loop as it can lead to performance issues and slow down the system.
- [ ]  When calling another function inside a given function, add comments explaining the purpose and meaning of the passed arguments and expected return values.
- [ ]  If adding a blank argument in a function, add a comment explaining the reason for the blank argument.
- [ ]  Before submitting a pull request, do a self-review of your code to ensure there are no conflicts with the base branch and all comments have been addressed.
- [ ]  Before merging a pull request, it's important to have other team members review it to catch any potential errors or issues
- [ ]  To maintain code integrity, it's important to remove all related changes when removing code during a code review.
- [ ]  If new constants, endpoints, or utility functions are introduced, it is important to check if they already exist in the service to avoid any duplication.
- [ ]  Whenever a new environment variable is added to a service, it's important to ensure that the necessary changes are made to related files such as ".env.sample" and "envVariables.js" to maintain consistency and avoid errors. Additionally, the new environment variable should be added to the devops repository to ensure that it is properly documented and accessible to the team.
- [ ]  When adding a new function to a service, it is important to document it with relevant information such as the name, parameters, and return value in a consistent format across all services. Additionally, if there are any changes to the API response, ensure that the documentation in the controllers is updated accordingly. 
- [ ]  Write a clear and concise commit message that describes the changes made.
- [ ]  Maintain consistent function signature and code across all services when adding a function to multiple services. Implement changes to database models in all services that use the same model. 
- [ ]  Use only let and const. Do not use var.
- [ ]  Make common functions for repetitive code blocks.
- [ ]  Avoid uploading sensitive information such as secret tokens or passwords in pull requests to ensure data security.
- [ ]  Maintain consistent indentation and spacing throughout the code.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "shikshalokam",
-  "version": "7.0.0",
+  "version": "8.0.0",
```
Comment (ankitpws) [inline package.json:3]: Add release note for 8.0.0 @praveenKDass

File: generics/helpers/mathFunctions.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
-const math = require('mathjs')
+const { create, all } = require('mathjs')
```
Comment (ankitpws) [inline generics/helpers/mathFunctions.js:1]: we can access the create function using math.create why it is specifically imported @praveenKDass

**Review by ankitpws — COMMENTED** [review]
Added Comments

File: controllers/v1/programOperationsController.js [hunk 3]
```diff
@@ -691,6 +691,12 @@ module.exports = class ProgramOperations {
         return new Promise(async (resolve, reject) => {
             try {
 
+                if (!/^[a-zA-Z0-9_-]+$/.test(req.query.id)) {
```
Comment (ankitpws) [inline controllers/v1/programOperationsController.js:694]: why is this added ? @praveenKDass

File: release-notes/8.0.0.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    1. Log in to Jenkins.
+    2. Navigate to Dashboard -> AutoBuild -> StagingManual -> managed-learn -> ml-survey-service. OR for dev Navigate to Dashboard -> Build -> managed-learn -> ml-survey-service
+    3. Click on "Build with parameters" and provide the latest release tag in the field labeled "github_release_tag". Initiate the build process.
+    4. The build job will take approximately 5 minutes to complete.
+    5. After the job finishes, go to Dashboard -> Deploy -> staging -> managed-learn -> ml-survey-service. OR for dev go to Dashboard -> Deploy -> dev -> managed-learn -> ml-survey-service This job will be executed automatically in the dev environment. If not, then it should be built manually.
+    6. Click on "Build with parameters" to initiate the deployment process.
+    7. Once the job is completed, the services will be deployed on the staging environment
+
+### No New Environment Keys Added
```
Comment (ankitpws) [inline release-notes/8.0.0.md:19]: not needed @praveenKDass

**Review by ankitpws — CHANGES_REQUESTED** [review]
Reviewed

File: controllers/v1/programOperationsController.js (hunk 3 above)
Comment (praveenKDass) [reply controllers/v1/programOperationsController.js:694]: This is a unfixed security issue added for sanitise the qurey.id. For Regular Expression Denial of Service (ReDoS)
issue

File: package.json (hunk 1 above)
Comment (praveenKDass) [reply package.json:3]: Added

File: controllers/v1/programOperationsController.js [hunk 5]
```diff
@@ -455,7 +455,7 @@ module.exports = class ProgramOperations {
 
                 assessorQueryObject["solutionId"] = ObjectId(req.params._id);
 
-                if (req.query.assessorName) assessorQueryObject["name"] = new RegExp(req.query.assessorName, 'i');
+                if (req.query.assessorName) assessorQueryObject["name"] = new RegExp(req.query.assessorName.replace(/[^a-zA-Z0-9_-]/g, ''), 'i');
```
Comment (ankitpws) [inline controllers/v1/programOperationsController.js:458]: Revert this change @praveenKDass

**Review by ankitpws — APPROVED** [review]
Apporved
